osu.Game/Screens/SelectV2/BeatmapCarousel.cs

```diff
@@ -176,39 +179,46 @@ private void beatmapSetsChanged(object? beatmaps, NotifyCollectionChangedEventAr
 
         protected override void HandleItemActivated(CarouselItem item)
         {
-            switch (item.Model)
+            try
             {
-                case GroupDefinition group:
-                    // Special case – collapsing an open group.
-                    if (lastSelectedGroup == group)
-                    {
-                        setExpansionStateOfGroup(lastSelectedGroup, false);
-                        lastSelectedGroup = null;
+                switch (item.Model)
+                {
+                    case GroupDefinition group:
+                        // Special case – collapsing an open group.
+                        if (lastSelectedGroup == group)
+                        {
+                            setExpansionStateOfGroup(lastSelectedGroup, false);
+                            lastSelectedGroup = null;
+                            return;
+                        }
+
+                        setExpandedGroup(group);
                         return;
-                    }
 
-                    setExpandedGroup(group);
-                    return;
+                    case BeatmapSetInfo setInfo:
+                        // Selecting a set isn't valid – let's re-select the first visible difficulty.
+                        if (grouping.SetItems.TryGetValue(setInfo, out var items))
+                        {
+                            var beatmaps = items.Select(i => i.Model).OfType<BeatmapInfo>();
+                            RequestRecommendedSelection(beatmaps);
+                        }
 
-                case BeatmapSetInfo setInfo:
-                    // Selecting a set isn't valid – let's re-select the first visible difficulty.
-                    if (grouping.SetItems.TryGetValue(setInfo, out var items))
-                    {
-                        var beatmaps = items.Select(i => i.Model).OfType<BeatmapInfo>();
-                        RequestRecommendedSelection(beatmaps);
-                    }
+                        return;
 
-                    return;
+                    case BeatmapInfo beatmapInfo:
+                        if (CurrentSelection != null && CheckModelEquality(CurrentSelection, beatmapInfo))
+                        {
+                            RequestPresentBeatmap?.Invoke(beatmapInfo);
+                            return;
+                        }
 
-                case BeatmapInfo beatmapInfo:
-                    if (CurrentSelection != null && CheckModelEquality(CurrentSelection, beatmapInfo))
-                    {
-                        RequestPresentBeatmap?.Invoke(beatmapInfo);
+                        RequestSelection(beatmapInfo);
                         return;
-                    }
-
-                    RequestSelection(beatmapInfo);
-                    return;
+                }
+            }
+            finally
+            {
+                playActivationSound(item);
```

Comment:
finally because we want to know the expanded state of groups to play the correct sample.